src/utils/AssayGeneratorMetabo.cpp

```diff
@@ -309,246 +214,398 @@ class TOPPAssayGeneratorMetabo :
+                                                             out_csifingerid,
+                                                             sirius_algo);
+  
+        // sort vector path list
+        std::sort(subdirs.begin(), subdirs.end(), extractAndCompareScanIndexLess_);
+        LOG_DEBUG << subdirs.size() << " spectra were annotated using SIRIUS." << std::endl;
+  
+        // get Sirius FragmentAnnotion from subdirs
+        vector<MSSpectrum> annotated_spectra;
+        for (auto subdir : subdirs)
         {
-          const MSSpectrum &spectrum = spectra[*index_it];
-          const BinnedSpectrum binned_spectrum(spectrum, BinnedSpectrum::DEFAULT_BIN_WIDTH_HIRES, false, 1, BinnedSpectrum::DEFAULT_BIN_OFFSET_HIRES);
-
-          BinnedSpectralContrastAngle bspa;
-          double cosine_sim = bspa(binned_highest_int, binned_spectrum);
-
-          if (cosine_sim > cosine_sim_threshold)
+          MSSpectrum annotated_spectrum;
+          SiriusFragmentAnnotation::extractSiriusFragmentAnnotationMapping(subdir, 
+                                                                           annotated_spectrum, 
+                                                                           use_exact_mass);
+          annotated_spectra.push_back(annotated_spectrum);
+        }
+        
+        // clean tmp directory if debug level < 2 
+        if (debug_level_ >= 2)
+        {
+          writeDebug_("Keeping temporary files in directory '" + tmp_dir + " and msfile at this location "+ tmp_ms_file + ". Set debug level to 1 or lower to remove them.", 2);
+        }
+        else
+        {
+          if (tmp_dir.empty() == false)
           {
-            similar_spectra.push_back(spectrum);
-            exp.addSpectrum(spectrum);
+            writeDebug_("Deleting temporary directory '" + tmp_dir + "'. Set debug level to 2 or higher to keep it.", 0);
+            File::removeDir(tmp_dir.toQString());
+          }
+          if (tmp_ms_file.empty() == false)
+          {
+            writeDebug_("Deleting temporary msfile '" + tmp_ms_file + "'. Set debug level to 2 or higher to keep it.", 0);
+            File::remove(tmp_ms_file); // remove msfile
           }
         }
-        // calculate consensus spectrum
-        exp.sortSpectra();
-        SpectraMerger merger;
-        Param p;
-        p.setValue("precursor_method:mz_tolerance", precursor_mz_distance);
-        p.setValue("precursor_method:rt_tolerance", precursor_rt_tol*2);
-        merger.setParameters(p);
-
-        // all MS spectra should have the same precursor
-        merger.mergeSpectraPrecursors(exp);
-
-        // check if all precursors have been merged if not use highest intensity precursor
-        if (exp.getSpectra().size() < 2)
+
+        // pair compoundInfo and fragment annotation msspectrum (using the mid)
+        for (auto cmp : v_cmpinfo)
         {
-          transition_spectrum = exp.getSpectra()[0];
+          for (auto spec_fa : annotated_spectra)
```

Comment:
```suggestion
          for (const auto& spec_fa : annotated_spectra)
```
unless you want to copy?